shallow_backup/utils.py

```diff
@@ -12,24 +12,30 @@ def run_cmd(command):
 	"""
 	try:
 		if not isinstance(command, list):
-			process = sp.run(command.split(), stdout=sp.PIPE)
+			process = sp.run(command.split(), stdout=sp.PIPE, stderr=sp.DEVNULL)
```

Comment:
I think we can do better than just telling the user something went wrong. 

What if we save the `stderr` output to a variable, and if there is any, let the user know there may have been an error and prompt Y/N to display the traceback/error?

Reply:
I think the streams need to be passed a file object. I got `AttributeError: 'str' object has no attribute 'fileno'` when trying to pass a variable.

If you want though, we could create a temporary file for `stderr`, write the traceback/error to it, then read its contents into a variable but that seems uh... gross lol. Plus the I/O overhead isn't doing us any favors.

Another option is instead of the actual traceback since that's not looking too hot, we can pass in the command that was run in *backup.py* and if the command's return code is nonzero, we prompt Y/N to display the command.

Reply:
> If you want though, we could create a temporary file for stderr, write the traceback/error to it, read its contents into a variable, then delete the file but that seems uh... gross lol. Plus the I/O overhead isn't doing us any favors.

From a logging perspective, it's preferable to have the original error, but I see what you're saying about the overhead.

> Another option is instead of the actual traceback since that's not looking too hot, we can pass in the command that was run in backup.py and if the command's return code is nonzero, we prompt Y/N to display the command.

I like this. If the command's return code is nonzero, let's print "An error occurred while running: $ _______".

Reply:
Do you still want to prompt y/n before that?

Reply:
No need to prompt, imo.

Reply:
3783008ac84fb8d1ec1d813b7ee9c2e50e0a8cc3

Reply:
I've been thinking about this some more and I'm a bit concerned about hiding all the stderr output. If I were a user and the shell commands fail, I'd like them to fail loudly and give me the error readout right in the terminal. I like handling errors for `shallow-backup` cleanly, but this would be hiding non-`shallow-backup` errors.

Does that make sense to you?

Reply:
just seeing this now. I know we resolved this further down and I have an edit down there sort of explaining my reasoning but in case you didn't see it:

Most of the package commands are just using the command's list functionality so it's not very likely that they'll fail (unless of course you don't have them installed which should be pretty obvious even with out basic error message). The only ones that I would be worried about are the packages that use `ls` instead of a native list option. Those could fail if a certain directory doesn't exist or something but at that point, failing loudly isn't that much more helpful since an `ls` error can only be of like 1 of 2 reasons lol.